example/memcached-operator/controller/resource/services.go

```diff
@@ -2,39 +2,179 @@ package resource
 
 import (
 	"context"
+	"fmt"
 
 	examplev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/example/memcached-operator/controller/key"
 	"github.com/giantswarm/operatorkit/example/memcached-operator/logger"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
 )
 
 const (
 	servicesName = "services"
 )
 
 type ServicesConfig struct {
+	K8sClient kubernetes.Interface
 }
 
 type Services struct {
+	k8sClient kubernetes.Interface
 }
 
 func NewServices(config ServicesConfig) (*Services, error) {
-	s := &Services{}
-	return s, nil
+	d := &Services{
+		k8sClient: config.K8sClient,
+	}
+
+	return d, nil
 }
-func (s *Services) Name() string {
+func (d *Services) Name() string {
 	return servicesName
 }
 
-func (s *Services) EnsureCreated(ctx context.Context, obj interface{}) error {
-	_ = obj.(*examplev1alpha1.MemcachedConfig)
+func (d *Services) EnsureCreated(ctx context.Context, obj interface{}) error {
+	memcachedConfig := obj.(*examplev1alpha1.MemcachedConfig).DeepCopy()
+
+	managedServices, err := d.k8sClient.CoreV1().Services(memcachedConfig.Namespace).List(metav1.ListOptions{
+		LabelSelector: key.LabelSelectorManagedBy,
+	})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	currentReplicas := len(managedServices.Items)
+	desiredReplicas := memcachedConfig.Spec.Replicas
+
+	logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("currentReplicas = %d desiredReplicas = %d", currentReplicas, desiredReplicas))
```

Comment:
This kind of logging is really not nice IMO. We have structured logging and here we then try to put structure into a string. Like why?